### PR TITLE
[SPARK-7801] [BUILD] Updating versions to SPARK 1.5.0

### PR DESCRIPTION
**Pull request by pwendell**

### CODE REVIEW

**Comment by pwendell** [comment]
Jenkins, test this please.

**Comment by pwendell** [comment]
Jenkins, test this please.

**Comment by pwendell** [comment]
Jenkins, retest this please.

**Comment by pwendell** [comment]
Jenkins, retest this please.

**Comment by pwendell** [comment]
Jenkins, retest this please.

**Comment by pwendell** [comment]
Jenkins, test this please.

**Comment by pwendell** [comment]
This is a pretty straightforward change so I'm just going to pull it in.

**Comment by pwendell** [comment]
Jenkins, retest this please (actually one more just to make sure MIMA is still passing)

**Comment by vanzin** [comment]
@pwendell any update on this? It would be nice to get into the custom of updating the master version early (usually when the release branch is created).

**Comment by pwendell** [comment]
@vanzin just updated to the newest RC. I kept racing with changes that were getting back-ported, so I gave up until the branch stabilized a bit.

We can do it earlier but then it becomes a burden for all new patches, since anything that does a refactoring in a way that removes API's (even from RCX -> RC[X + 1]) has to add a MIMA exception. It turns out we do this kind of often.

**Comment by pwendell** [comment]
BTW - this is also important to do soon because of the fact that we publish snapshot releases now from both master and 1.4 branches.

**Comment by pwendell** [comment]
Jenkins, retest this please.

**Comment by srowen** [comment]
Jenkins, retest this please

File: project/MimaExcludes.scala [hunk 1]
```diff
@@ -34,6 +34,20 @@ import com.typesafe.tools.mima.core.ProblemFilters._
 object MimaExcludes {
     def excludes(version: String) =
       version match {
+        case v if v.startsWith("1.5") =>
+          Seq(
+            MimaBuild.excludeSparkPackage("deploy"),
+            // These are needed if checking against the sbt build, since they are part of
+            // the maven-generated artifacts in 1.3.
+            excludePackage("org.spark-project.jetty"),
```
Comment (vanzin) [inline project/MimaExcludes.scala]: Not sure this is strictly needed since 1.4 has the shaded jetty already, but doesn't hurt either.

**Comment by vanzin** [comment]
LGTM.

File: project/MimaExcludes.scala (hunk 1 above)
Comment (pwendell) [inline project/MimaExcludes.scala]: It was actually failing before I added it. Probably worth digging into, but given all the races with this patch I am inclined to maybe merge this in now and then worry about it later.

**Pull request closed without merging**